### DISCUSSION TITLE
[RAWeb API] Login endpoint

### DISCUSSION THREAD

**Discussion by WedgeSparda** [RetroAchievements/RAWeb#2841] [Ideas]
As far as I know, the only ways to login into are using the website (obviously) and the rcheevos cliient.

I’m trying to create a login into RA for an iOS app in order to get the API Key for the user. So I can…

A) Show a webview from the app and capture the html response to scrap the key.
B) Somehow I integrate rcheevos client in my app (despite not being an emulator) and use to retrieve the key.

Are these my only options? Can we get a Web API endpoint for this purpose or does it present security issues?

**Comment by wescopeland** [comment]
Ideally, we facilitate this by supporting OAuth2 in RAWeb.

**Comment by WedgeSparda** [comment]
So, it’s not supported right now?

**Reply by wescopeland** [reply]
It is not currently supported. Integrating OAuth2 is on our roadmap, but will require a complete rewrite of how authentication is currently handled in the application via a migration to Laravel Passport.
